packages/services/f-test-data-creator/src/clients/publicApi.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable camelcase */
+const axios = require('axios');
+const { getLanguageForTenant } = require('../configuration/tenants');
+
+module.exports = class PublicApiService {
+    constructor (configuration) {
+        this.createUserUrl = configuration.Services.Public.CreateUserUrl;
+        this.authorizationUrl = configuration.Services.Public.AuthorizationUrl;
+        this.token = configuration.Services.Public.AuthorizationToken;
+    }
+
+    async createConsumer (userInfo, timeout = 5000) {
+        const config = {
+            headers: {
+                'Content-Type': 'application/json',
+                Authorization: this.token
+            },
+            timeout
+        };
+
+        const requestData = {
+            registrationSource: 'Native',
+            firstName: userInfo.firstName,
+            lastName: userInfo.lastName,
+            password: userInfo.password,
+            emailAddress: userInfo.emailAddress,
+            tenant: userInfo.tenant
+        };
+
+        return axios.post(this.createUserUrl, requestData, config)
+        .catch(error => {
+            throw new Error(error.message);
+        });
+    }
+
+    async getAuthCodeUk (emailAddress, password, timeout = 5000) {
+        const config = {
+            headers: {
+                Authorization: this.token
+            },
+            timeout
+        };
+
+        const params = {
+            acr_values: 'tenant:uk language:en-GB',
+            scope: 'openid mobile_scope offline_access',
+            grant_type: 'password',
+            username: emailAddress,
+            password
+        };
+
+        const searchParams = Object.keys(params).map(key => `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}`).join('&');
+
+        return axios.post(this.authorizationUrl, searchParams, config)
+        .catch(error => {
+            throw new Error(error.message);
+        });
+    }
+
+    async getAuthCodeInt (emailAddress, password, tenant, timeout = 5000) {
+        const config = {
+            headers: {
+                Authorization: this.token
+            },
+            timeout
+        };
+
+        const params = {
+            acr_values: `tenant:${tenant} language:${getLanguageForTenant(tenant)}`,
+            scope: 'openid mobile_scope offline_access',
+            grant_type: 'password',
+            username: emailAddress,
+            password
+        };
+
+        const searchParams = Object.keys(params).map(key => `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}`).join('&');
```

Comment:
Might be worth leaving a comment on why we're doing this.

Reply:
yup will do